src/main/scala/nl.knaw.dans.easy.multideposit/actions/ValidatePreconditions.scala

```diff
@@ -46,10 +46,10 @@ class ValidatePreconditions(ldap: Ldap) extends DebugEnhancedLogging {
       .getOrElse(Success(()))
   }
 
-  def checkOutputDirectoryExists(depositId: DepositId)(implicit output: OutputPathExplorer): Try[Unit] = {
+  def checkOutputDirectoryExists(bagId: BagId, depositId: DepositId)(implicit output: OutputPathExplorer): Try[Unit] = {
```

Comment:
does the error message still make sense?

Reply:
The addition of the bagId is because the name of the 'staged bag' (while building the bag) is different than the one from the 'output bag'. This `bagId` (in fact a `UUID`) is assigned by the tool itself, not by the user of the tool. Therefore the user doesn't even know about the `bagId` and its link with the data he/she just deposited. Thus, to indicate that the deposit failed, we need to show the `depositId` instead, which is chosen by the user!

However, now that I think of this somewhat more, I'm not even sure we still need this check. After all, we need to validate that the output directory does not yet exists, but the name is this output directory is the `bagId` (`UUID`) and is therefore (semi-)unique. The chance that this check fails is very minimal and if it fails it is not an error caused by the user. In fact, maybe we should not have this check here at all, but move it to the place where the `bagId` is decided:

```
IF the generated bagId already exists as the name of a directory in the output directory
THEN we need to generate a new bagId and repeat this check
ELSE return the current bagId
```

@jo-pol @lindareijnhoudt @janvanmansum does this make sense and should we implement it this way?

Reply:
If we can't trust a UUID to be unique, we'll need such a check anywhere we generate a UUID. Thinking about that, it makes sense to add such a check everywhere we generate a UUID. For example for the [deposit-api](https://github.com/DANS-KNAW/easy-deposit-api/blob/14d05b6d1f937088c753f90c1401315f116cc6fb/src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala#L257-L259)

Reply:
I see what you mean, but if you [read this](https://stackoverflow.com/a/1155027/2389405) you understand why I'm questioning whether this is necessary.

> the annual risk of a given person being hit by a meteorite is estimated to be one chance in 17 billion, which means the probability is about 0.00000000006 (6 × 10−11), equivalent to the odds of creating a few tens of trillions of UUIDs in a year and having one duplicate. In other words, only after generating 1 billion UUIDs every second for the next 100 years, the probability of creating just one duplicate would be about 50%.